pyscriptjs/tests/integration/support.py

```diff
@@ -92,6 +93,32 @@ def init(self, request, tmpdir, http_server, logger, page):
 
     def init_page(self, page):
         self.page = page
+        # cache to store data via sha256(url)
+        cache = {}
+
+        # router to cache with fail 2x on requests
+
+        def router(route):
+            # hash of url
+            hash = hashlib.sha256(route.request.url.encode("utf-8")).hexdigest()
+            # cached?
+            if hash in cache:
+                # fulfill via cache
+                route.fulfill(status=200, response=cache.get(hash))
+            else:
+                try:
+                    # fetch, cache, fulfill 1x
+                    response = page.request.fetch(route.request)
+                    cache[hash] = response
+                    route.fulfill(status=200, response=response)
+                except Exception:
```

Comment:
this is dangerous, I'd prefer to catch only the specific kind of exception that we expect

Reply:
Discovering the Exception types that page.request.fetch generates. Will narrow this down tight.

Reply:
- Added PlaywrightRequestError to isolate
- Support for --headed